GVFS/GVFS.Common/ProductUpgrader.Shared.cs

```diff
@@ -14,17 +14,27 @@ public partial class ProductUpgrader
         private const string GVFSInstallerFileNamePrefix = "SetupGVFS";
         private const string VFSForGitInstallerFileNamePrefix = "VFSForGit";
 
-        public static bool IsLocalUpgradeAvailable()
+        public static bool IsLocalUpgradeAvailable(string[] installerExtensions)
         {
             string downloadDirectory = GetAssetDownloadsPath();
             if (Directory.Exists(downloadDirectory))
             {
-                const string PotentialInstallerName = "*VFS*.*";
-                string[] installers = Directory.GetFiles(
-                    downloadDirectory,
-                    PotentialInstallerName, 
-                    SearchOption.TopDirectoryOnly);
-                return installers.Length > 0;
+                HashSet<string> extensions = new HashSet<string>(installerExtensions, StringComparer.OrdinalIgnoreCase);
+                HashSet<string> installerNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase)

```

Comment:
Same here, why do we need a hashset for this? With the current logic, this will return true if there is a git installer but not a GVFS installer. Is that desired?

If we're ok with partial results, then I would make this a single check for the GVFS installer name.

Reply:
Current logic- Check for the files "SetupGVFS.<version>.exe" or "VFSForGit.<version>.exe". Return true if one of these GVFS installers is found. Both these files are GVFS installers (in the new and old name format), we don't check for Git installer.

We need a Collection class (HashSet or List) only because of the need to support both current and new GVFS installer names in future.